### ISSUE TITLE
[Bug]: `aws_route53_record` fails when `ttl` is set to `0`

### ISSUE THREAD

**Issue by danielmotaleite** [hashicorp/terraform-provider-aws#36411]
### Terraform Core Version

v1.4.6

### AWS Provider Version

v5.40.0

### Affected Resource(s)

aws_route53_record

### Expected Behavior

When ttl = 0 (zero), the resource should be created, just in line as the AWS route53 console

### Actual Behavior

Error, no resource created, while the plan show OK
Can't recreate via terraform a record created manually in the AWS console

### Relevant Error/Panic Output Snippet

```shell
Error: creating Route 53 Record: InvalidInput: Invalid request: Expected exactly one of [AliasTarget, all of [TTL, and ResourceRecords], or TrafficPolicyInstanceId], but found none in Change with [Action=UPSERT, Name=replica-1.example.com, Type=CNAME, SetIdentifier=replica-1]
```

### Terraform Configuration Files

```
  resource "aws_route53_record" "this" {
       allow_overwrite = true
       name            = "replica-1.example.com"
       records         = [
           "staging-1.zzzzzzz.rds.amazonaws.com",
        ]
       set_identifier  = "replica-1"
       ttl             = 0
       type            = "CNAME"
       zone_id         = "XXXX"
    }

```

### Steps to Reproduce

Create the tf  to create a route53 record, with TTL  = 0 
try to apply the terraform
it should fail.
change the TTL> 0
apply the terraform
it should work now

### Important Factoids

The AWS console didn't allow TTL in the past, but now it do allows and TTL= 0 is usefull for internal zones, as a weak load balancer, namely for RDS replica with healthchecks (route53_healthchecks), to add or remove replicas based on several checks. For that, we do not want cached results, as that may negate the healtcheck objective

### Would you like to implement a fix?

No

**Comment by hugo-barros** [comment]
I went to tackle this one (as I was looking for a simple issue to be my first contribution), but my acceptance test didn't pass (with same error). I wonder whether it is possible to achieve this without the plugin framework? (Instead of sdkv2).

**Comment by danielmotaleite** [comment]
@hugo-barros thank you for your effort
@justinretzolk any change of helping hugo in this problem

**Comment by danielmotaleite** [comment]
just to add that this makes usage of  weighted DNS extremely annoying, as it force me to manual change the ttl from zero to 1 to add any new entry, and them manually reset them back to zero

`│ Error: creating Route 53 Record: InvalidChangeBatch: [RRSet with DNS name readreplica-amc.staging.internal., type CNAME, SetIdentifier readreplica-amc-4 cannot be created as weighted sets must contain the same TTL., RRSet with DNS name readreplica-amc.staging.internal., type CNAME, SetIdentifier readreplica-amc-4 cannot be created as weighted sets must contain the same TTL.]`

**Comment by danielmotaleite** [comment]
i upgraded to  v5.70. and this still exists in the latest version

**Comment by danielmotaleite** [comment]
confirmed fixed with 5.73.0, i can create/update records with TTL=0
